pkg/sources/kafka/reader.go

```diff
@@ -254,11 +254,16 @@ func (r *KafkaSource) Pending(ctx context.Context) (int64, error) {
 		return isb.PendingNotAvailable, fmt.Errorf("failed to list consumer group offsets, %w", err)
 	}
 	for _, partition := range partitions {
+		block := rep.GetBlock(r.topic, partition)
+		if block.Offset == -1 {
+			// This partition has not been processed yet.
```

Comment:
This comment is still misleading, update it together with some other PRs.

Reply:
I thought it was valid, thinking what the comment meant was this partition had never been processed; hence it is -1. If it ever read something, it wouldn't be -1.

@KeranYang is this understanding correct?

Reply:
@vigith Yes your understanding is correct. I think the word `process` can be misleading as people can think that `not been processed`  means the partition has data but not being consumed yet. I can add more details and explicitly mention the partition has no data.